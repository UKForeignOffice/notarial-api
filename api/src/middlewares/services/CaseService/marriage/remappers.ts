import * as affirmationMappings from "./mappings/affirmation";
import * as cniPostalMappings from "./mappings/cniPostal";
import * as exchangeMappings from "./mappings/exchange";
import * as mscMappings from "./mappings/msc";
import * as cniMSCCombinedMappings from "./mappings/CNIAndMSC";
import { createRemapper } from "../utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
  cni: createRemapper(affirmationMappings.remap, [], ["file"]),
  cniPostal: createRemapper(cniPostalMappings.remap, [], ["file"]),
  exchange: createRemapper(exchangeMappings.remap, [], ["file"]),
  mscPostal: createRemapper(mscMappings.remap, [], ["file"]),
  cniAndMscPostal: createRemapper(cniMSCCombinedMappings.remap, [], ["file"]),
};
