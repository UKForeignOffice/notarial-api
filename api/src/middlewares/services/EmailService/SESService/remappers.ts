import * as affirmationMappings from "./mappings/affirmation";
import { createRemapper } from "./utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
};
