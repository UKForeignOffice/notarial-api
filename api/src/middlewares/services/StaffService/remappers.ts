import * as affirmationMappings from "./mappings/affirmation";
import * as exchangeMappings from "./mappings/exchange";
import { createRemapper } from "./utils/createRemapper";

export const remappers = {
  affirmation: createRemapper(affirmationMappings.remap, [], ["file"]),
  cni: createRemapper(affirmationMappings.remap, [], ["file"]),
  exchange: createRemapper(exchangeMappings.remap, [], ["file"]),
};
