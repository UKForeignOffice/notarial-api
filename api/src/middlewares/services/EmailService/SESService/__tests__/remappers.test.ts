import { remappers } from "../remappers";
import { fields } from "./fixtures/fields";

test("affirmation is successfully remapped", () => {
  const toAffirmation = remappers.toAffirmation;
  const remapped = fields.reduce(toAffirmation, {});
  expect(remapped).toMatchSnapshot();
});
