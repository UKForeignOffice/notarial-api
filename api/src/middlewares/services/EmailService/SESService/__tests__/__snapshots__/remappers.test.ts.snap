// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`affirmation is successfully remapped 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantBirth": {
    "countryOfBirth": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryOfBirth",
      "title": "Country of birth",
      "type": "list",
    },
    "nameChanged": {
      "answer": "By deed poll",
      "category": "nameChange",
      "key": "nameChanged",
      "title": "Name changed",
      "type": "list",
    },
    "placeOfBirth": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "placeOfBirth",
      "title": "Place of birth ",
      "type": "text",
    },
  },
  "applicantDetails": {
    "dateOfBirth": {
      "answer": "2000-11-11",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "emailAddress": {
      "answer": "pye@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+440000000",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "sex": {
      "answer": "Female",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "applicantPassport": {
    "dateOfIssue": {
      "answer": "2020-12-12",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "0000",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "applicationBirth": {
    "previousName": {
      "answer": "Testing",
      "category": "nameChange",
      "key": "previousName",
      "title": "Previous name",
      "type": "text",
    },
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": true,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "certifyPassport": {
      "answer": true,
      "category": "beforeYouStart",
      "key": "certifyPassport",
      "title": "Do you want to certify your passport as part of your affirmation application?",
      "type": "list",
    },
    "country": {
      "answer": "Belarus",
      "category": "beforeYouStart",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
    "jurats": {
      "answer": [
        "you have no additional reading or writing needs",
      ],
      "category": "oath",
      "key": "UPQLxm",
      "title": "Select all that apply to youÂ ",
      "type": "list",
    },
    "oathType": {
      "answer": "Non-religious",
      "category": "oath",
      "key": "oathType",
      "title": "Type of oath",
      "type": "list",
    },
  },
  "maritalStatus": {
    "dateOfDecreeAbsolute": {
      "answer": "2020-12-12",
      "category": "maritalStatus",
      "key": "dateOfDecreeAbsolute",
      "title": "Date of issue of decree absolute",
      "type": "date",
    },
    "placeOfDecreeAbsolute": {
      "answer": "London, England",
      "category": "maritalStatus",
      "key": "placeOfDecreeAbsolute",
      "title": "Place of issue of decree absolute",
      "type": "text",
    },
    "status": {
      "answer": "Divorced",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2024-06",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "Test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "occupation": {
    "occupation": {
      "answer": "Test",
      "category": "occupation",
      "key": "occupation",
      "title": "Occupation",
      "type": "text",
    },
  },
  "parentDetails": {
    "fatherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "fatherFullName",
      "title": "Father's full name",
      "type": "text",
    },
    "motherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "motherFullName",
      "title": "Mother's full name",
      "type": "text",
    },
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "Spain",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner's first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner's middle name",
      "type": "text",
    },
    "partnerSurname": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner's surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "United States citizen",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner's nationality",
      "type": "list",
    },
  },
}
`;

exports[`affirmation remapper 1`] = `
{
  "applicantAddress": {
    "addressLine1": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "applicantBirth": {
    "countryOfBirth": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryOfBirth",
      "title": "Country of birth",
      "type": "list",
    },
    "nameChanged": {
      "answer": "By deed poll",
      "category": "nameChange",
      "key": "nameChanged",
      "title": "Name changed",
      "type": "list",
    },
    "placeOfBirth": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "placeOfBirth",
      "title": "Place of birth ",
      "type": "text",
    },
  },
  "applicantDetails": {
    "dateOfBirth": {
      "answer": "2000-11-11",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "emailAddress": {
      "answer": "pye@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+440000000",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "sex": {
      "answer": "Female",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "applicantPassport": {
    "dateOfIssue": {
      "answer": "2020-12-12",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "0000",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "applicationBirth": {
    "previousName": {
      "answer": "Testing",
      "category": "nameChange",
      "key": "previousName",
      "title": "Previous name",
      "type": "text",
    },
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": true,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "certifyPassport": {
      "answer": true,
      "category": "beforeYouStart",
      "key": "certifyPassport",
      "title": "Do you want to certify your passport as part of your affirmation application?",
      "type": "list",
    },
    "country": {
      "answer": "Belarus",
      "category": "beforeYouStart",
      "key": "country",
      "title": "Country",
      "type": "list",
    },
    "jurats": {
      "answer": [
        "you have no additional reading or writing needs",
      ],
      "category": "oath",
      "key": "UPQLxm",
      "title": "Select all that apply to youÂ ",
      "type": "list",
    },
    "oathType": {
      "answer": "Non-religious",
      "category": "oath",
      "key": "oathType",
      "title": "Type of oath",
      "type": "list",
    },
  },
  "maritalStatus": {
    "dateOfDecreeAbsolute": {
      "answer": "2020-12-12",
      "category": "maritalStatus",
      "key": "dateOfDecreeAbsolute",
      "title": "Date of issue of decree absolute",
      "type": "date",
    },
    "placeOfDecreeAbsolute": {
      "answer": "London, England",
      "category": "maritalStatus",
      "key": "placeOfDecreeAbsolute",
      "title": "Place of issue of decree absolute",
      "type": "text",
    },
    "status": {
      "answer": "Divorced",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "marriageDetails": {
    "dateOfMarriage": {
      "answer": "2024-06",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "Test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "occupation": {
    "occupation": {
      "answer": "Test",
      "category": "occupation",
      "key": "occupation",
      "title": "Occupation",
      "type": "text",
    },
  },
  "parentDetails": {
    "fatherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "fatherFullName",
      "title": "Father's full name",
      "type": "text",
    },
    "motherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "motherFullName",
      "title": "Mother's full name",
      "type": "text",
    },
  },
  "partnerAddress": {
    "partnerAddressCity": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "Spain",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "partnerName": {
    "partnerFirstName": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner's first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner's middle name",
      "type": "text",
    },
    "partnerSurname": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner's surname",
      "type": "text",
    },
  },
  "partnerNationality": {
    "partnerNationality": {
      "answer": "United States citizen",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner's nationality",
      "type": "list",
    },
  },
}
`;
